client/my-sites/domains/domain-management/transfer/transfer-to-other-site/index.jsx

```diff
@@ -183,13 +218,18 @@ export class TransferToOtherSite extends React.Component {
 }
 
 export default connect(
-	( state, ownProps ) => ( {
-		currentUser: getCurrentUser( state ),
-		domainsWithPlansOnly: currentUserHasFlag( state, DOMAINS_WITH_PLANS_ONLY ),
-		isDomainOnly: isDomainOnlySite( state, get( ownProps, 'selectedSite.ID', null ) ),
-		sites: getSites( state ),
-		currentRoute: getCurrentRoute( state ),
-	} ),
+	( state, ownProps ) => {
+		const domain = ! ownProps.isRequestingSiteDomains && getSelectedDomain( ownProps );
```

Comment:
@hambai, would love your opinion here - seems that when the cache is not rehydrated in dev mode (meaning, we start with empty cache), the `isRequestingSiteDomains` flag does not seem to work as it should 🤔 Meaning, I was expecting a loading placeholder (see above), but seems that there's a fatal error instead because `currentUserCanManage` is resolved as false-y.
I'm not sure if I'm missing something obvious here - I know our data layer for domains is a bit spotty, so maybe I'm doing something wrong (though, tried following existing patterns).

Reply:
I am not in a position to dive in yet, but I am eager to learn more about what may be going awry here.

As a complete stab in the dark, it may be worth trying to access the state directly in this component, if only to confirm whether more direct access affects the behaviour in any way.

Reply:
Well, looks like this fixed itself - I remember seeing a related chatter in #calypso last week about someone else having a similar issue. I've rebased the patch and now I cannot reproduce it anymore (even with forcing cache re-hydration: `?flags=force-sympathy`

Reply:
Nope, looks like it's still an issue:

```
Warning: Failed prop type: The prop `currentUserCanManage` is marked as required in `TransferToOtherSite`, but its value is `undefined`.
    in TransferToOtherSite (created by Localized(TransferToOtherSite))
    in Localized(TransferToOtherSite) (created by ConnectFunction)
    in ConnectFunction (created by StoreConnection)
    in StoreConnection (created by DomainManagementData)
    in div (created by DomainManagementData)
    in DomainManagementData (created by ConnectFunction)
    in ConnectFunction
```

Reply:
Phew, OK, I figured it out: https://github.com/Automattic/wp-calypso/pull/46418/commits/6cfa145b63dcc228345b5fbf383eb1907fe72425
Looks like we have two flags that have similar names but different meanings: `isRequestingSiteDomains` and `hasSiteDomainsLoaded`. First means we are loading new data - but does not say if already have (cached) data loaded. That's what the other one says. And we need both to properly check if we have any data to display.